frontend/src/components/object/ObjectTable.vue

```diff
@@ -197,6 +216,8 @@ watch( selectedObjects, () => {
             <font-awesome-icon icon="fa-solid fa-users" />
           </Button>
           <Button
+            v-if="permissionStore.isObjectActionAllowed(
+              data.id, getUserId, Permissions.READ, props.bucketId as string) || data.public"
```

Comment:
It's better to evaluate obj.public first to leverage short circuiting; that way we can save some computation if it's already public and avoid needing to hit the permissionStore.